pkgs/development/libraries/libcef/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv, mkChromiumDerivation, fetchgit, gtkglext
+, fontconfig, freetype, cairo, pango, libX11, libXi, libXext
+, libXfixes, libXrender, libXcomposite, dbus, atk, libXrandr
+, libXmu, libXt, libSM, libICE }:
+
+with stdenv.lib;
+
+mkChromiumDerivation (base: rec {
+  name = "libcef";
+  packageName = "cef";
+  # Run "git rev-list --count HEAD" to get this number
+  cefVersion = "1414";
+  # from CHROMIUM_BUILD_COMPATIBILITY.txt
+  upstreamVersion = "51.0.2704.47";
+  version = assert base.version == upstreamVersion;
+            "${upstreamVersion}-${cefVersion}";
+
+  buildTargets = [ "mksnapshot" "chrome_sandbox" "libcef" ];
+
+  buildInputs = base.buildInputs ++ [ gtkglext ];
+  rpath = buildInputs ++ [
+    fontconfig freetype cairo pango libX11 libXi libXext libXfixes
+    libXrender libXcomposite dbus atk libXrandr stdenv.cc.cc
+  ];
+
+  cefSrc = fetchgit {
+    url = "https://bitbucket.org/chromiumembedded/cef.git";
+    rev = "185cd6c507a018593d310742fa5c7fa3adc53cb4";
+    sha256 = "0g2y1fwp0cvid9xlgz0ixgbxz7ryqiwpfac2l58sc3zdgf2iayjd";
+  };
+
+  postUnpack = ''
+    cp -prvd "${cefSrc}" "$sourceRoot/cef"
+    chmod -R u+w "$sourceRoot/cef"
+  '';
+
+  preConfigure = base.preConfigure + ''
+    cd cef
+
+    substituteInPlace tools/make_version_header.py \
+      --replace "git.is_checkout('.')" "True" \
+      --replace "git.get_commit_number()" '"${cefVersion}"' \
+      --replace "git.get_hash()" '"${cefSrc.rev}"'
+    substituteInPlace tools/commit_number.py \
+      --replace "git.is_checkout('.')" "True" \
+      --replace "git.get_commit_number()" '"${cefVersion}"'
+
+    python tools/make_version_header.py \
+      --header include/cef_version.h \
+      --cef_version VERSION \
+      --chrome_version ../chrome/VERSION \
+      --cpp_header_dir include
+
+    python tools/patcher.py \
+      --patch-config patch/patch.cfg
+  '';
+
+  configureScript = "python ../build/gyp_chromium";
+  configureFlags = "cef.gyp -I cef.gypi --depth .. ${base.gypFlags}";
```

Comment:
Actually I did it in this very way, see [this line](https://github.com/NixOS/nixpkgs/pull/15648/commits/4b4acf94045a277b484839efd028a36f194ac007#diff-fdf71de5b0eb5fb927a9706511410f93R67). I'm still `cd`ing back and forth but only in `preConfigure` -- to type smaller paths.